### DISCUSSION TITLE
splitChunks behavior is affected by cacheGroups.name field

### DISCUSSION THREAD

**Discussion by JSerFeng** [webpack/webpack#17704] [Q&A]
## Example

Say we have a module relationship like this:

```
index --dyn import()--> a --import--> x, y
index --dyn import()--> b --import--> x
```

Before splitChunks, we have 2 chunks here, one for `a` and one for `b`.

We set cacheGroups.default.text to `/x|y/`, so `x` and `y` both match, and will be split later (assume that we match the minSize here).

The default strategy of splitChunks, will split `x` in a common chunk , which is expected as it shows up in 2 chunks. And `y` still only appear in chunk `a`.

But if we set cacheGroups.default.name to any value, `x` and `y` appear in the same chunk. If `x` and `y` are css modules, this behavior will cause some priority problems.

![image](https://github.com/webpack/webpack/assets/57202839/7dd9834b-ae86-4a71-b55c-7e5a7d77805d)

## Code Sandbox

https://codesandbox.io/p/sandbox/bold-orla-5xqm5d?file=%2Foutput%2Findex.js%3A12%2C19

run `npm run build`, see the result, and comment or uncomment cacheGroups.default.name, see the differences

## Questions

I know the reason in source code, its because the key of `chunkInfoMap` is affected by name if specified, otherwise it is based on the chunks combinations.

But I do not know if this is by design or a bug, personally I feel confused if name affects splitChunks results

**Comment by alexander-akait** [comment]
Sorry for delay, you can use:
```
default: {
          type: /javascript/,
          name: "defaultChunk",
          test: /x|y/,
          minChunks: 1,
        },
```

Otherwise webpack tries to split all possible chunks, inclused CSS stuff.

> But I do not know if this is by design or a bug, personally I feel confused if name affects splitChunks results

That is interesting question, because both examples make sense...

I am fine with dicussion about it
